bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/handler/PentairIPBridgeHandler.java

```diff
@@ -31,86 +31,57 @@
  * @author Jeff James - Initial contribution
  *
  */
+
+@NonNullByDefault
 public class PentairIPBridgeHandler extends PentairBaseBridgeHandler {
     private final Logger logger = LoggerFactory.getLogger(PentairIPBridgeHandler.class);
 
+    public PentairIPBridgeConfig config = new PentairIPBridgeConfig();
+
     /** Socket object for connection */
-    protected Socket socket;
+    protected @Nullable Socket socket;
 
     public PentairIPBridgeHandler(Bridge bridge) {
         super(bridge);
     }
 
     @Override
-    protected synchronized void connect() {
-        PentairIPBridgeConfig configuration = getConfigAs(PentairIPBridgeConfig.class);
+    protected synchronized int connect() {
```

Comment:
Instead of returning an `int` it would be better to throw exceptions here instead. Or if you only care about connection success or failure then returning a boolean would also work.

Reply:
just used a boolean so i didn't have to change structure of code.